aws/provider.go

```diff
@@ -1072,18 +1072,21 @@ func assumeRoleSchema() *schema.Schema {
 					Type:        schema.TypeString,
 					Optional:    true,
 					Description: descriptions["assume_role_role_arn"],
+					DefaultFunc: schema.EnvDefaultFunc("AWS_ASSUME_ROLE_ARN", nil),
 				},
 
 				"session_name": {
 					Type:        schema.TypeString,
 					Optional:    true,
 					Description: descriptions["assume_role_session_name"],
+					DefaultFunc: schema.EnvDefaultFunc("AWS_ASSUME_SESSION_NAME", nil),
 				},
 
 				"external_id": {
 					Type:        schema.TypeString,
 					Optional:    true,
 					Description: descriptions["assume_role_external_id"],
+					DefaultFunc: schema.EnvDefaultFunc("AWS_ASSUME_EXTERNAL_ID", nil),
```

Comment:
Maybe `AWS_ASSUME_ROLE_EXTERNAL_ID` to match the AWS code property name?
https://github.com/terraform-providers/terraform-provider-aws/blob/905ba44ccd6b3e442f709878abadf458e114f0cc/aws/provider.go#L1023-L1025